src/v/storage/segment_index.cc

```diff
@@ -156,8 +181,9 @@ ss::future<> segment_index::truncate(model::offset o) {
 }
 
 ss::future<bool> segment_index::materialize_index() {
-    auto size = co_await _out.size();
-    auto buf = co_await _out.dma_read_bulk<char>(0, size);
+    auto f = co_await open();
```

Comment:
We don't close this file `f`.

Reply:
Fixed.

Reply:
@jcsp should I be seeing a new push, or these fixes are pending?

Reply:
> We don't close this file f.

marked fixed, but don't see it closed. leaving note here since it might be done in a later patch

Reply:
ok now i see it